pkg/controller/controller.go

```diff
@@ -567,6 +572,10 @@ func (p *csiProvisioner) ProvisionExt(options controller.ProvisionOptions) (*v1.
 		return nil, controller.ProvisioningFinished, fmt.Errorf("failed to strip CSI Parameters of prefixed keys: %v", err)
 	}
 
+	// add pvc metadata to request for use by the plugin
+	req.Parameters[pvcNameKey] = options.PVC.GetName()
```

Comment:
we should gate this behind a feature flag or binary flag of some sort. This is a breaking change for drivers that do validation on parameters and don't expect extra stuff coming in. (this should default to off and you can turn it on if you want this information)

Reply:
It seems like the featureFlag gate used here is being pulled from the api server, i'm not too sure if i should extend it with custom provisioner-only feature gates or just add a boolean flag say `--inject-pvc-metadata`, what do you think is the best approach?

Reply:
I think a boolean flag is fine. I suck at naming so I can't come up with another solution but I don't think it should contain `pvc` in it since we will inject `pv/pvc` name. It should also have some sort of information of "where" it is injecting the metadata. `extra-create-params-metadata`?

Can you add a comment disclaimer as a flag comment along the lines of:
`This metadata should only be used for tagging or descriptive purposes and should in no circumstances be consumed and acted on programatically by a CSI Driver.`

Reply:
Are we going to follow some sort of Alpha->Beta->GA plan for this flag?
If so can we include the stage in the name '--extra-create-params=alpha-pvc'?

Reply:
no need for a stage - just a single flag should be fine

Reply:
What I am worried about in that case is that we will keep adding flags and the command to run the container will be really large by default.

It would be nice to have this be the default behavior at some point and the flag would be used to turn it off.

Reply:
```      --feature-gates mapStringBool            A set of key=value pairs that describe feature gates for alpha/experimental features. Options are:
                                               AllAlpha=true|false (ALPHA - default=false)
                                               AllBeta=true|false (BETA - default=false)
                                               Topology=true|false (BETA - default=false)
                                               enable-extra-metadata-for-create-req=true|false (ALPHA - default=false)
```
I added this feature flag, does this sound reasonable? I accept name suggestions btw...

Reply:
Or do you prefer
```
 --enable-extra-metadata-for-create-req   If set, add pv/pvc metadata to plugin create requests as parameters.
```
I have both implement, just let me know which makes more sense.

Reply:
I personally prefer the feature gate and have it go to default at some point.

Reply:
we can't set this as a default unless we bump the major version of the provisioner - it is a breaking change to drivers as this is adding "unexpected" parameters to their CreateVolume API call. I don't really see the value of feature gating this as the intention for that is to slowly "graduate" features to minimize risk. I wouldn't really consider this a risky feature in terms of potential impact or implementation

Reply:
Ok, i'll add a standard flag for this, any suggestions on naming? Anything else needing change? I couldn't find any place to document this feature for consumption...

Reply:
Also open an issue to track changing it to default, and we can add it to the 2.0 milestone